python/lsst/ip/isr/fringe.py

```diff
@@ -104,14 +104,41 @@ def readFringes(self, dataRef, assembler=None):
             fringe = dataRef.get("fringe", immediate=True)
         except Exception as e:
             raise RuntimeError("Unable to retrieve fringe for %s: %s" % (dataRef.dataId, e))
+
+        return self.loadFringes(fringe, assembler)
+
+    def loadFringes(self, fringeExp, assembler=None):
+        """Pack the fringe data into a Struct.
+
+        This method moves the struct parsing code into a butler
+        generation agnostic handler.
+
+        Parameters
+        ----------
+        fringeExp : `lsst.afw.exposure.Exposure`
+            The exposure containing the fringe data.
+        assembler : `lsst.ip.isr.AssembleCcdTask`, optional
+            An instance of AssembleCcdTask (for assembling fringe
+            frames).
+
+        Returns
+        -------
+        fringeData : `pipeBase.Struct`
+            Struct containing fringe data:
+            - ``fringes`` : `lsst.afw.image.Exposure` or `list` thereof
+                Calibration fringe files containing master fringe frames.
+            - ``seed`` : `int`, optional
+                Seed for random number generation.
+        """
         if assembler is not None:
-            fringe = assembler.assembleCcd(fringe)
+            fringeExp = assembler.assembleCcd(fringeExp)
+
+        seed = self.config.stats.rngSeedOffset + 12345678
```

Comment:
why this magic number, if no reason add a comment. If there is a reason, add a comment

Reply:
Updated to try to use the exposureId as in gen2 instead of a magic number.